### PR TITLE
Sanity check socc and docc arrays to avoid segfaults.

### PR DESCRIPTION
**Pull request by susilehtola**
## Description
The user-supplied occupation arrays weren't sanity checked, which caused segfaults.

This PR makes psi4 give an error message instead.

## Status
- [x]  Ready to go

### CODE REVIEW

File: psi4/src/psi4/libscf_solver/hf.cc [hunk 1]
```diff
@@ -153,18 +153,18 @@ void HF::common_init()
         if(old_pg){
             // This is one of a series of displacements;  check the dimension against the parent point group
             size_t full_nirreps = old_pg->char_table().nirrep();
-            if(options_["DOCC"].size() != full_nirreps)
-                throw PSIEXCEPTION("Input DOCC array has the wrong dimensions");
```
Comment (dgasmith) [inline psi4/src/psi4/libscf_solver/hf.cc]: This is still an important check. Other than that, I think everything looks good.

Comment (susilehtola) [inline psi4/src/psi4/libscf_solver/hf.cc]: Good catch!

**Comment by susilehtola** [comment]
I just used the same fork that I used for the earlier commits. I'm still kinda new to PRs in github... am I supposed to delete the repo and refork every time for a new PR?

**Comment by amjames** [comment]
@susilehtola it is no problem to use the same fork without deleting/re-forking over and over. 
You would usually set up a new branch for your work, push that branch to your fork, and then open the PR. Then once the PR is merged you can delete that branch

The general setup I normally follow goes something like this. 


- Bring my master branch (in my local psi4 repo) up to date with the master branch on psi4/psi4 (upstream)
```
>>> git checkout master
>>> git pull --rebase upstream master
```
- Create a new branch (locally) to make my changes on
```
>>> git checkout -b fix_this_bug
>>> … edit the offending files …
>>> git add <offending files>
>>> git commit -m "<code> no longer does <something it shouldn't> fixes #<issue>" <— that issue will be automatically closed when the PR gets merged
```
- Now I can push that branch to my fork (origin)
```
>>> git push origin fix_this_bug
         this will push my local branch "fix_this_bug" 
         to the branch "fix_this_bug" 
          on the remote origin (my fork)
```
- When I go to my fork's page I see a prompt asking if I want to open a PR since  I just pushed a new branch. Click it, fill out the PR and it's open!

I just cloned the PR locally and tried to do a rebase, but It failed due to conflicts. Not sure what to do here @dgasmith might know how to sort it out. 

Usually if your PR falls behind (ie some other PR gets merged) you can just do another 
`git pull --rebase upstream master`, then `git push —force  origin fix_this_bug` but in this case it looks like that might make things worse.

**Comment by dgasmith** [comment]
What happened was your previous commits were rebased so your local and master git hashes differed. However, since the new commits had the same diff as your current commits they effectively did nothing. Probably not terribly clear unless your familiar with git.

The best thing to do as AJ mentioned is to always make a new branch a pull down a fresh version of `psi4/psi4:master`.  

@amjames The histories probably were muddled. We probably need to squash this anyway so we can make a merge as the diff looks fine.

**Pull request merged**